gnomad_hail/utils/annotations.py

```diff
@@ -419,3 +425,503 @@ def annotate_freq(
+        }
+    )
+
+    trio_stats = trio_stats.annotate(
+        **{
+            f'ac_parents_{name}': hl.agg.filter(
+                expr,
+                hl.agg.sum(trio_mt.father_entry.GT.n_alt_alleles() + trio_mt.mother_entry.GT.n_alt_alleles())
+            ) for name, expr in ac_strata.items()
+        },
+        **{
+            f'ac_children_{name}': hl.agg.filter(
+                expr,
+                hl.agg.sum(trio_mt.proband_entry.GT.n_alt_alleles())
+            ) for name, expr in ac_strata.items()
+        }
+    )
+
+    return trio_stats
+
+
+def filter_mt_to_trios(
+        mt: hl.MatrixTable,
+        fam_ht: hl.Table
+) -> hl.MatrixTable:
+    """
+    Filters a MatrixTable to a set of trios in `fam_ht`, filters to autosomes, and annotates with adj.
+
+    :param mt: A Matrix Table to filter to only trios
+    :param fam_ht: A Table of trios to filter to, loaded using `hl.import_fam`
+    :return: A MT filtered to trios and adj annotated
+    """
+    # Filter MT to samples present in any of the trios
+    fam_ht = fam_ht.annotate(
+        fam_members=[fam_ht.id, fam_ht.pat_id, fam_ht.mat_id]
+    )
+    fam_ht = fam_ht.explode('fam_members', name='s')
+    fam_ht = fam_ht.key_by('s').select().distinct()
+
+    mt = mt.filter_cols(hl.is_defined(fam_ht[mt.col_key]))
+    mt = filter_to_autosomes(mt)
+    mt = annotate_adj(mt)
+
+    return mt
+
+
+def default_generate_trio_stats(
+        mt: hl.MatrixTable,
+        ped: hl.Pedigree
+    ) -> hl.Table:
+    """
+    Default function to run `generate_trio_stats_expr` to get trio stats stratified by raw and adj
+
+    .. note::
+
+        Expects that `mt` is annotated with adj and if dealing with a sparse MT,
+        `hl.experimental.densify` must be run first.
+
+    :param mt: A Matrix Table of only trios
+    :param ped: A Pedigree of trios to calculates stats on loaded using `hl.Pedigree.read`
+    :return: Table with trio stats
+    """
+    mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+    mt = hl.trio_matrix(mt, pedigree=ped, complete_trios=True)
+    logger.info(f"Generating trio stats using {mt.count_cols()} trios.")
+    trio_adj = (mt.proband_entry.adj & mt.father_entry.adj & mt.mother_entry.adj)
+
+    ht = mt.select_rows(
+        **generate_trio_stats_expr(
+            mt,
+            transmitted_strata={
+                'raw':  None,
+                'adj': trio_adj
+            },
+            de_novo_strata={
+                'raw': None,
+                'adj': trio_adj
+            },
+            ac_strata={
+                'raw': True,
+                'adj': trio_adj
+            },
+            proband_is_female_expr=mt.is_female
+        )
+    ).rows()
+
+    return ht.filter(
+        ht.n_de_novos_raw + ht.n_transmitted_raw + ht.n_untransmitted_raw > 0
+    )
+
+
+def generate_sib_stats_expr(
+    sib_mt: hl.MatrixTable,
+    strata: Dict[str, hl.expr.BooleanExpression] = {"raw": True},
+    sib1_entry: str = 'sib1_entry',
```

Comment:
It's probably something that we need to address more widely, but passing MT + col names vs expr is something we should try and be consistent with. I now tend to like expr better because they can be nested and / or computed from other expr without needing an `annotate`. Anyhow, just a thought for something more global rather than immediate action needed.